src/main/java/net/sf/jabref/logic/layout/LayoutEntry.java

```diff
@@ -268,38 +261,37 @@ private String handleOptionField(BibEntry bibtex, BibDatabase database) {
     }
 
     private String handleFieldOrGroupStart(BibEntry bibtex, BibDatabase database, Optional<Pattern> highlightPattern) {
-        String field;
+        Optional<String> field;
         if (type == LayoutHelper.IS_GROUP_START) {
             field = BibDatabase.getResolvedField(text, bibtex, database);
         } else if (text.matches(".*(;|(\\&+)).*")) {
```

Comment:
A great improvement would be to compile the regexes to pattern and then use simply
split()  on that pattern object , otherwise java creates each regex pattern everytime again.
(I did not know that until I just found it in javadoc  :sunglasses: )
https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#split-java.lang.CharSequence-